### PR TITLE
feat: transient queries added to show queries output

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
Addresses https://github.com/confluentinc/ksql/issues/4308

There's a known issue where if no persistent queries are running in a multi node environment and a transient query is running on node A, that transient query won't show up in `Show queries` executed to other nodes.
https://github.com/confluentinc/ksql/issues/4910

### Testing done 
```
 Query ID                                                            | Query Type | Status    | Sink Name | Sink Kafka Topic | Query String                                                                                                                                  
---------------------------------------------------------------------------------------------------------------------------------------------
 CSAS_TEST_0                                                         | PUSH       | RUNNING:2 | TEST      | TEST             | CREATE STREAM TEST WITH (KAFKA_TOPIC='TEST', PARTITIONS=1, REPLICAS=1) AS SELECT * FROM KSQL_PROCESSING_LOG KSQL_PROCESSING_LOG EMIT CHANGES; 
 _confluent-ksql-default_transient_4363766092250723780_1587488499684 | TRANSIENT  | RUNNING:1 |           |                  | select * from test emit changes;                                                                                                              
---------------------------------------------------------------------------------------------------------------------------------------------
```

```
ID                   : _confluent-ksql-default_transient_4363766092250723780_1587488499684
Query Type           : TRANSIENT
SQL                  : select * from test emit changes;
Host Query Status    : {192.168.1.6:8089=RUNNING}

 Field   | Type                                                                                                                                                                                                                                                                                                                                    
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ROWTIME | BIGINT                                                                                                                                                                                                                                                                                                                                  
 ROWKEY  | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                         
 LOGGER  | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                         
 LEVEL   | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                         
 TIME    | BIGINT                                                                                                                                                                                                                                                                                                                                  
 MESSAGE | STRUCT<TYPE INTEGER, DESERIALIZATIONERROR STRUCT<ERRORMESSAGE VARCHAR(STRING), RECORDB64 VARCHAR(STRING), CAUSE ARRAY<VARCHAR(STRING)>, topic VARCHAR(STRING)>, RECORDPROCESSINGERROR STRUCT<ERRORMESSAGE VARCHAR(STRING), RECORD VARCHAR(STRING), CAUSE ARRAY<VARCHAR(STRING)>>, PRODUCTIONERROR STRUCT<ERRORMESSAGE VARCHAR(STRING)>> 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Sources that this query reads from: 
-----------------------------------
TEST

For source description please run: DESCRIBE [EXTENDED] <SourceId>

Execution plan      
--------------      
 > [ PROJECT ] | Schema: ROWKEY STRING KEY, ROWTIME BIGINT, ROWKEY STRING, LOGGER STRING, LEVEL STRING, TIME BIGINT, MESSAGE STRUCT<TYPE INTEGER, DESERIALIZATIONERROR STRUCT<ERRORMESSAGE STRING, RECORDB64 STRING, CAUSE ARRAY<STRING>, `topic` STRING>, RECORDPROCESSINGERROR STRUCT<ERRORMESSAGE STRING, RECORD STRING, CAUSE ARRAY<STRING>>, PRODUCTIONERROR STRUCT<ERRORMESSAGE STRING>> | Logger: 4363766092250723780.Project
                 > [ SOURCE ] | Schema: ROWKEY STRING KEY, LOGGER STRING, LEVEL STRING, TIME BIGINT, MESSAGE STRUCT<TYPE INTEGER, DESERIALIZATIONERROR STRUCT<ERRORMESSAGE STRING, RECORDB64 STRING, CAUSE ARRAY<STRING>, `topic` STRING>, RECORDPROCESSINGERROR STRUCT<ERRORMESSAGE STRING, RECORD STRING, CAUSE ARRAY<STRING>>, PRODUCTIONERROR STRUCT<ERRORMESSAGE STRING>>, ROWTIME BIGINT, ROWKEY STRING | Logger: 4363766092250723780.KsqlTopic.Source


Processing topology 
------------------- 
Topologies:
   Sub-topology: 0
    Source: KSTREAM-SOURCE-0000000000 (topics: [TEST])
      --> KSTREAM-TRANSFORMVALUES-0000000001
    Processor: KSTREAM-TRANSFORMVALUES-0000000001 (stores: [])
      --> Project
      <-- KSTREAM-SOURCE-0000000000
    Processor: Project (stores: [])
      --> KSTREAM-FOREACH-0000000003
      <-- KSTREAM-TRANSFORMVALUES-0000000001
    Processor: KSTREAM-FOREACH-0000000003 (stores: [])
      --> none
      <-- Project
```

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs-md/developer-guide/ksqldb-reference/show-queries.md [hunk 1]
```diff
@@ -30,15 +30,21 @@ Query Status
 * `ERROR`: the query has entered an error state.
 * `UNRESPONSIVE`: the host running the query returned an error when requesting the query status.
 
+Query Type
+-----------
+
+* `Push`: these queries run on every single node in the cluster.
```
Comment (JimGalasyn) [inline docs-md/developer-guide/ksqldb-reference/show-queries.md:36]:
```suggestion
* `Push`: these queries run on every node in the cluster.

```

File: docs-md/developer-guide/ksqldb-reference/show-queries.md [hunk 2]
```diff
@@ -30,15 +30,21 @@ Query Status
 * `ERROR`: the query has entered an error state.
 * `UNRESPONSIVE`: the host running the query returned an error when requesting the query status.
 
+Query Type
+-----------
```
Comment (agavra) [inline docs-md/developer-guide/ksqldb-reference/show-queries.md:34]: cc @AlanConfluent can you review this section? I think we're standardizing on:
- **persistent** queries run on every node and materialize new state
- **push** queries are owned by the client and are terminated when the session ends (I don't think we should document that they only run on a single node, that's not the key characteristic)

File: ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConstants.java [hunk 3]
```diff
@@ -46,6 +46,11 @@ private KsqlConstants() {
   public static final String DEFAULT_AVRO_SCHEMA_FULL_NAME =
       AVRO_SCHEMA_NAMESPACE + "." + AVRO_SCHEMA_NAME;
 
+  public enum KsqlQueryType {
```
Comment (agavra) [inline ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConstants.java:49]: see the documentation comment above

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java [hunk 4]
```diff
@@ -128,10 +130,16 @@ public int numberOfLiveQueries() {
...
   }
 
+  @Override
   public List<PersistentQueryMetadata> getPersistentQueries() {
     return ImmutableList.copyOf(primaryContext.getPersistentQueries().values());
   }
 
+  @Override
+  public List<QueryMetadata> getAllLiveQueries() {
+    return ImmutableList.copyOf(new ArrayList<>(allLiveQueries));
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java:140]:
```suggestion
    return ImmutableList.copyOf(allLiveQueries);
```
the `new ArrayList` is not necesasry, `copyOf` takes an `Iterable`

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java [hunk 5]
```diff
@@ -100,19 +100,36 @@ private ListQueriesExecutor() { }
...
+                return new RunningQuery(
+                    q.getStatementString(),
+                    ImmutableSet.of(persistentQuery.getSinkName().text()),
+                    ImmutableSet.of(persistentQuery.getResultTopic().getKafkaTopicName()),
+                    q.getQueryId(),
+                    QueryStatusCount.fromStreamsStateCounts(
+                        Collections.singletonMap(KafkaStreams.State.valueOf(q.getState()), 1)),
+                    q.getQueryType());
+              }
+              System.out.println("what is being returned");
```
Comment (agavra) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java:121]:
```suggestion
```

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java [hunk 6]
```diff
@@ -100,19 +100,36 @@ private ListQueriesExecutor() { }
...
-                q.getStatementString(),
-                ImmutableSet.of(q.getSinkName().text()),
-                ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
-                q.getQueryId(),
-                QueryStatusCount.fromStreamsStateCounts(
-                    Collections.singletonMap(KafkaStreams.State.valueOf(q.getState()), 1)))
+            QueryMetadata::getQueryId,
+            q -> {
+              if (q instanceof PersistentQueryMetadata) {
+                System.out.println("getting local");
```
Comment (agavra) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java:110]:
```suggestion
```

File: docs-md/developer-guide/ksqldb-reference/show-queries.md (hunk 2 above)
Comment (AlanConfluent) [reply docs-md/developer-guide/ksqldb-reference/show-queries.md]: Yes, I think the definitions you stated here are good ones.  The lifetime I think is one of the most important qualities from a client's perspective as well as the state materialized by persistent queries.  You could call out that persistent queries run until explicitly terminated, that might be too verbose for this purpose.

FYI, I started documenting terminology here, but it's still a work in progress given the changing push query direction so might not be fully up to date with current thinking: https://confluentinc.atlassian.net/wiki/spaces/KSQL/pages/1193742550/ksqlDB+Query+Terminology

**Comment by newsmallyu** [comment]
How can I manually stop these transient queries？

**Comment by newsmallyu** [comment]
I encountered the problem that I cannot stop these transient queries when I use rest  /query

**Comment by nateab** [comment]
> I encountered the problem that I cannot stop these transient queries when I use rest /query

Work is being done to fix this issue #5868

**Pull request merged**